src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationService.kt

```diff
@@ -575,14 +576,22 @@ class ValidationService(
 
   private fun validateDtoIsNotRecall(prisonApiSourceData: PrisonApiSourceData): List<ValidationMessage> {
     val validationMessages = mutableListOf<ValidationMessage>()
+    var bookingHasDto = false
+    var bookingHasRecall = false
     prisonApiSourceData.sentenceAndOffences.forEach() {
 
       val sentenceCalculationType = SentenceCalculationType.from(it.sentenceCalculationType)
-      val hasRecall = it.terms.any { terms ->
+      val hasDtoRecall = it.terms.any { terms ->
         terms.code == SentenceTerms.BREACH_OF_SUPERVISION_REQUIREMENTS_TERM_CODE || terms.code == SentenceTerms.BREACH_DUE_TO_IMPRISONABLE_OFFENCE_TERM_CODE
       }
       val hasDto = sentenceCalculationType == DTO || sentenceCalculationType == DTO_ORA
-      if (featureToggles.dto && hasDto && hasRecall) {
+      if (hasDto) {
+        bookingHasDto = true
+      }
+      if (sentenceCalculationType.recallType != null) {
+        bookingHasRecall = true
+      }
+      if (featureToggles.dto && hasDto && hasDtoRecall) {
```

Comment:
I think we may have a bit of a problem in the validation service. The validation messages should appear in a certain order. I think the unsupported calculation errors should be returned before `BREACH_DUE_TO_IMPRISONABLE_OFFENCE_TERM_CODE` type validations. Also the general DTO unsupported error I think needs to be returned before others.